core/frontend/src/store/popups.ts

```diff
@@ -0,0 +1,67 @@
+import Vue from 'vue'
+import {
+  Action, getModule, Module, Mutation, VuexModule,
+} from 'vuex-module-decorators'
+
+import store from '@/store'
+import { ActivePopup, PopupOptions, PopupResult } from '@/types/popups'
+
+@Module({
+  dynamic: true,
+  store,
+  name: 'popups',
+})
+class PopupsStore extends VuexModule {
+  popups: Record<string, ActivePopup> = {}
+
+  @Mutation
+  addPopup(popup: ActivePopup): void {
+    Vue.set(this.popups, popup.id, popup)
+  }
+
+  @Mutation
+  removePopup(id: string): void {
+    Vue.delete(this.popups, id)
+  }
+
+  @Mutation
+  close(id: string): void {
+    if (this.popups[id]) {
+      this.popups[id].dismissed = true
+      this.popups = { ...this.popups }
+    }
+  }
+
+  @Mutation
+  closeAll(): void {
+    Object.keys(this.popups).forEach((id) => {
+      this.popups[id].dismissed = true
+    })
+
+    this.popups = { ...this.popups }
+  }
+
+  @Action
+  fire(options: PopupOptions): Promise<PopupResult> {
+    return new Promise<PopupResult>((resolve) => {
+      this.addPopup({
+        id: options.id as string,
```

Comment:
The ID is essentially a string. However, the popup options provide users with the flexibility to leave it unset because it is an optional property. If the user does not set it, the fire method within the Popup functionality automatically assigns a value to it. Consequently, when this data is passed to the addPopup method, it is treated as an ActivePopup. Within this context the ID is already defined either by user or fire method, so the conversion is only to remove the assignment from string | undefined from the PopupOption to the ActivePopup.

We could also change it on the Action fire to receive a direct id parameter that is already string and that will be defined in the fire method, in this way the compiler can check that is not undefined